streams/src/main/java/org/apache/kafka/streams/processor/internals/ProcessorStateManager.java

```diff
@@ -262,9 +265,14 @@ public StateStore getStore(final String name) {
         return registeredStores.getOrDefault(name, Optional.empty()).orElse(null);
     }
 
+    /**
+     * @throws ProcessorStateException error when flushing the state store; for example IO exception
```

Comment:
My intent is to separate out `ProcessorStateException` and `TaskMigratedException` from other `StreamsException` (these two are extending from it); but thinking about it I think only `TaskMigratedException` needs to be separate out since `ProcessorStateException` should be considered fatal just like other `StreamsException` (as you can see, we just wrap a non-StreamsException as a ProcessorStateException, e.g. an IOException when flushing the store, and other StreamsException are re-thrown as is).

So I'm happy to just remove the `ProcessorStateException` if you prefer, just lmk.

Reply:
Sure, I suppose that would clarify it, but if no one is going to handle a `ProcessorStateException` specially, we can just directly throw a `StreamsException` and save some complexity.

As for whether you update the comment, or take this further step or not, it's up to you. I just found the javadocs confusing and felt it worth mentioning, since the point of this is to make everything less confusing.

Reply:
I guess I will remove this now.